react/admin/CustomField.tsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react'
+import { Input } from 'vtex.styleguide'
+
+import type { CustomField } from './CustomFields'
+
+interface CustomFieldProps {
+  index: number
+  handleUpdate: (index: number, customField: CustomField) => void
+  fieldLabel: string
+  fieldValue: string
+  fieldType: 'text'
```

Comment:
I think I'd pass in the entire `CustomField` object as a prop instead of using these custom props. In particular I don't like that we're duplicating `text` as the type all over the place, that is going to make it annoying if we want to add new field types.

Reply:
✅ Added a `CustomField` interface